Doc(next): add documentation for integrating Shadcn Admin Kit with Next (#10)
marmelab/shadcn-admin-kit: docs/using-with-next.md

```

## Configure Typescript

Once the new project has been created, we have to update the `tsconfig.app.json` as described in this repository's [`README.md`](../README.md).

```json
{
  "compilerOptions": {
    // ...
    "verbatimModuleSyntax": false
  }
}
```

## Install Shadcn Admin Kit Components

Once our project has been initialized, we can pull the `shadcn-admin-kit` components using the `shadcn` command:

```bash
npx shadcn@latest add https://marmelab.com/shadcn-admin-kit/r/shadcn-admin-kit-base.json
```

This will create install the `Shadcn` components inside the `components/ui` directory, the `Shadcn Admin Kit` insite the `components/admin` directory, and utilities inside the `hooks` and `lib` directories.

| Admin components                                               | Utilities                                                     |
| -------------------------------------------------------------- | ------------------------------------------------------------- |
| ![](./images/shadcn-admin-kit-next/source-code-components.png) | ![](./images/shadcn-admin-kit-next/source-code-utilities.png) |

## Configure the Admin App

> The `components/admin/admin.tsx` component supports the same data providers as the traditional [React Admin version](https://marmelab.com/react-admin/DataProviders.html). The list of [supported backends](https://marmelab.com/react-admin/DataProviderList.html) is available on the React Admin documentation. In this example, we will use the `ra-data-json-server` with fake data for the admin.

Reviewer: ```suggestion
The `<Admin>` component (`components/admin/admin.tsx`) requires a **dataProvider** to communicate with you API. We'll use a third-party package, `ra-data-json-server` to map the [JSONPlaceholder](https://jsonplaceholder.typicode.com/) API to the admin CRUD API. There are [dozens of data provider packages](./DataProviderList.md) for various APIs and databases. You can also create your own if necessary. For now, let's make sure the app connects to JSONPlaceholder.
```

> First, we will create the `components/AdminApp.tsx` file that will contain our admin app. we use the installed `ListGuesser` and `EditGuessers` from `Shadcn Admin Kit` in our two resources.

Reviewer: I'd keep the `AdminApp.tsx` file out of `components`. At least that's what we did in the demo. To me the `components` folder is rather intended to hold reusable components, not application pages.

Also:

```suggestion
First, we will create the `components/AdminApp.tsx` file that will contain our admin app. We will leverage the `ListGuesser` and `EditGuesser` components provided by `Shadcn Admin Kit` in our two resources.
```

```tsx
// components/AdminApp.tsx

"use client"; // remove this line if you choose Pages Router
import { Resource } from "ra-core";

> import jsonServerProvider from "ra-data-json-server";

Reviewer: You say nowhere to actually install the `ra-data-json-server` package. I think that's worth mentioning.

import { Admin, EditGuesser, ListGuesser } from "./admin";

const dataProvider = jsonServerProvider("https://jsonplaceholder.typicode.com");

const AdminApp = () => (
  <Admin dataProvider={dataProvider}>
    <Resource
      name="users"
      list={ListGuesser}
      edit={EditGuesser}

>       recordRepresentation="name"

Reviewer: You could add a `**Tip:**` pointing to the `recordRepresentation` doc https://marmelab.com/react-admin/Resource.html#recordrepresentation

    />
    <Resource
      name="posts"
      list={ListGuesser}
      edit={EditGuesser}
      recordRepresentation="title"
    />
    <Resource name="comments" list={ListGuesser} edit={EditGuesser} />
  </Admin>
);

export default AdminApp;
```

> Then we can create the `components/Admin.tsx` by using the same code as the [React Admin Next.js](https://marmelab.com/react-admin/NextJs.html) integration tutorial.

Reviewer: ```suggestion
Then we can create a `components/Admin.tsx` file to dynamically import our admin component and render it with SSR disabled (since it's a SPA).
```

```tsx
// components/Admin.tsx

"use client";
import dynamic from "next/dynamic";

const Admin = dynamic(() => import("./AdminApp"), {
  ssr: false, // Required to avoid react-router related errors
});

export default Admin;
```

## Create the Admin Page

Once the components have been configured, you can create the admin page at `app/admin/page.tsx`:

```tsx
// app/admin/page.tsx

> import Admin from "@/components/Admin";

Reviewer: This is very confusing, compared to imports from `"@/components/admin"`. You should find another name.

export default function Page() {
  return <Admin />;
}
```

Then, run `npm run dev` and go to `http://localhost:3000/admin` to access your admin app!

> ![](./images/shadcn-admin-kit-next/demo.png)

Reviewer: Why not provide an alt text? :flushed: 

Besides, this screenshot is a bit too big to fit on a doc page, not even talking about mobile.
Can you make another one on a smaller screen?

);

export default AdminApp;

```

**Tip:** You can use the  [`recordrepresentation`](https://marmelab.com/react-admin/Resource.html#recordrepresentation) with resources to configure how the record will be rendered by Shadcn Admin Kit.

## Create the Admin Page

Once the <AdminApp /> has been configured, we can create the admin page at `app/admin/page.tsx` to dynamically import our admin component and render it with SSR disabled (since it's a SPA).

```tsx

> // app/admin/page.tsx

Reviewer 2: It's not missing, we use a dynamic import so it's not on top of the file, but see line 117.

> // app/admin/page.tsx

Reviewer: Indeed

"use client";

import dynamic from "next/dynamic";

const Admin = dynamic(() => import("./AdminApp"), {
  ssr: false, // Required to avoid react-router related errors
});

export default function Page() {
  return <Admin />;
}

```

Then, run `npm run dev` and go to `http://localhost:3000/admin` to access your admin app!

> ![A Demo Admin Build with Shadcn Admin Kit](./images/shadcn-admin-kit-next/demo.jpg)

Reviewer: Build -> Built

> ![A Demo Admin Build with Shadcn Admin Kit](./images/shadcn-admin-kit-next/demo.jpg)

Reviewer: the screenshot is still huge, can't you take it from a smaller window? :thinking:
